ui/app/AppLayouts/Onboarding2/OnboardingLayout.qml

```diff
@@ -0,0 +1,559 @@
+    }
+
+    enum PrimaryPath {
+        Unknown,
+        CreateProfile,
+        Login
+    }
+
+    enum SecondaryPath {
+        Unknown,
+
+        CreateProfileWithPassword,
+        CreateProfileWithSeedphrase,
+        CreateProfileWithKeycard,
+        CreateProfileWithKeycardNewSeedphrase,
+        CreateProfileWithKeycardExistingSeedphrase,
+
+        LoginWithSeedphrase,
+        LoginWithSyncing,
+        LoginWithKeycard
+    }
+
+    // page stack
+    StackView {
+        id: stack
+        anchors.fill: parent
+        initialItem: welcomePage
+
+        pushEnter: Transition {
+            ParallelAnimation {
+                NumberAnimation { property: "opacity"; from: 0; to: 1; duration: d.opacityDuration; easing.type: Easing.InQuint }
+                NumberAnimation { property: "x"; from: (stack.mirrored ? -0.3 : 0.3) * stack.width; to: 0; duration: d.swipeDuration; easing.type: Easing.OutCubic }
+            }
+        }
+        pushExit: Transition {
+            NumberAnimation { property: "opacity"; from: 1; to: 0; duration: d.opacityDuration; easing.type: Easing.OutQuint }
+        }
+        popEnter: Transition {
+            ParallelAnimation {
+                NumberAnimation { property: "opacity"; from: 0; to: 1; duration: d.opacityDuration; easing.type: Easing.InQuint }
+                NumberAnimation { property: "x"; from: (stack.mirrored ? -0.3 : 0.3) * -stack.width; to: 0; duration: d.swipeDuration; easing.type: Easing.OutCubic }
+            }
+        }
+        popExit: pushExit
+        replaceEnter: pushEnter
+        replaceExit: pushExit
+    }
+
+    MouseArea {
+        anchors.fill: parent
+        acceptedButtons: Qt.BackButton
+        enabled: stack.depth > 1 && !stack.busy
+        cursorShape: undefined // fall thru
+        onClicked: stack.pop()
+    }
+
+    StatusBackButton {
+        width: 44
+        height: 44
+        anchors.left: parent.left
+        anchors.leftMargin: Theme.padding
+        anchors.bottom: parent.bottom
+        anchors.bottomMargin: Theme.padding
+        opacity: stack.depth > 1 && !stack.busy ? 1 : 0
+        visible: opacity > 0
+        Behavior on opacity { NumberAnimation { duration: 100 } }
+        onClicked: stack.pop()
+    }
+
+    // main signal handler
+    Connections {
+        id: mainHandler
+        target: stack.currentItem
+        ignoreUnknownSignals: true
+
+        // common popups
+        function onPrivacyPolicyRequested() {
+            console.warn("!!! AUX: PRIVACY POLICY")
+            privacyPolicyPopup.createObject(root).open()
+        }
+        function onTermsOfUseRequested() {
+            console.warn("!!! AUX: TERMS OF USE")
+            termsOfUsePopup.createObject(root).open()
+        }
+        function onOpenLink(link: string) {
+            Global.openLink(link)
+        }
+        function onOpenLinkWithConfirmation(link: string, domain: string) {
+            Global.openLinkWithConfirmation(link, domain)
+        }
```

Comment:
Those should be handled uniformly across the app; notice this happens at one place only here in the OnboardingLayout.qml; I could as well re-emit that signal so that it gets handled in AppMain